app/src/main/java/dev/hossain/weatheralert/di/WeatherRepositoryBindings.kt

```diff
@@ -0,0 +1,23 @@
+package dev.hossain.weatheralert.di
+
+import dev.hossain.weatheralert.data.ApiKeyProvider
+import dev.hossain.weatheralert.data.ApiKeyProviderImpl
+import dev.hossain.weatheralert.data.WeatherRepository
+import dev.hossain.weatheralert.data.WeatherRepositoryImpl
+import dev.zacsweers.metro.AppScope
+import dev.zacsweers.metro.BindingContainer
+import dev.zacsweers.metro.ContributesTo
+import dev.zacsweers.metro.Provides
+import dev.zacsweers.metro.SingleIn
+
+@ContributesTo(AppScope::class)
+@BindingContainer
+object WeatherRepositoryBindings {
+    @Provides
+    @SingleIn(AppScope::class)
+    fun provideWeatherRepository(impl: WeatherRepositoryImpl): WeatherRepository = impl
+
+    @Provides
+    @SingleIn(AppScope::class)
+    fun provideApiKeyProvider(impl: ApiKeyProviderImpl): ApiKeyProvider = impl
+}
```

Comment:
This file was a mistake and wrong. Fixed in https://github.com/hossain-khan/android-weather-alert/pull/453 using contributes binding.